src/messages/MOSDSubOp.h

```diff
@@ -194,7 +198,6 @@ class MOSDSubOp : public Message {
     ::encode(old_size, payload);
     ::encode(old_version, payload);
     ::encode(snapset, payload);
-    ::encode(snapc, payload);
```

Comment:
This will break.  If we are talking to an older peer who only understands v11, we needt o send a message that includes an encoded (dummy) snapc.  That means we need to use a feature bit and connditionally encode an empty snapc here if the peer doesn't have it (and set head.version = 11 instead of HEAD_VERSION).

Feature bits are getting scarce, but we are about to merge a bunch of stuff that will need one, so we can piggyback them all on the same bit.  Go ahead and define one (include/ceph_features.h) and we'll merge them together when merging.  For example, there is another pull request that makes the MOSDSubOp[Reply] message do partial encoding.. see https://github.com/ceph/ceph/pull/6503